File: Graph/ScriptTemplate/IsNil/CGisNil.js
/**
 * @file CGisNil.js
 * @author
 * @date 2021/8/15
 * @brief CGisNil.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGisNil extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    if (this.inputs[0]() == null) {
      return true;
    } else {
      return false;
    }
  }
}

exports.CGisNil = CGisNil;

File: Graph/ScriptTemplate/IsNil/CGIsNil.lua
local CGIsNil = CGIsNil or {}
CGIsNil.__index = CGIsNil

function CGIsNil.new()
    local self = setmetatable({}, CGIsNil)
    self.inputs = {}
    return self
end

function CGIsNil:setInput(index, func)
    self.inputs[index] = func
end

function CGIsNil:getOutput(index)
    if self.inputs[0]() == nil then
        return true
    else
        return false
    end
end

return CGIsNil
